funcbench/main.go

```diff
@@ -0,0 +1,356 @@
+	if err != nil {
+		return nil, err
+	}
+	raceArgument, err := readFile("/github/home/commentMonitor/RACE")
+	home := os.Getenv("HOME")
+
+	bench := benchmarkTester{
+		raceFlagEnabled:          raceArgument != "-no-race",
+		home:                     home,
+		bechRegex:                bechRegex,
+		comparisonBranchFilePath: strings.Join([]string{home, "/old.txt"}, ""),
+		pullRequestFilePath:      strings.Join([]string{home, "/new.txt"}, ""),
+	}
+	log.Printf("Benchmark Tester: %+v\n", bench)
+	return &bench, err
+}
+
+func (bench *benchmarkTester) compareBenchmarks() (string, error) {
+	b, err := exec.Command(strings.Join([]string{os.Getenv("GOPATH"), "/bin/benchcmp"}, ""), bench.comparisonBranchFilePath, bench.pullRequestFilePath).CombinedOutput()
+	log.Println(string(b))
+	return string(b), err
+}
+
+func (bench *benchmarkTester) execBenchmark() ([]byte, error) {
+	var cmd *exec.Cmd
+	if bench.raceFlagEnabled {
+		cmd = exec.Command("go", "test", "-bench", fmt.Sprintf("^%s$", bench.bechRegex), "-benchmem", "-v", "./...")
+	} else {
+		cmd = exec.Command("go", "test", "-bench", fmt.Sprintf("^%s$", bench.bechRegex), "-benchmem", "-race", "-v", "./...")
+	}
+	data, err := cmd.CombinedOutput()
+	log.Println(string(data))
+	if err != nil || cmd.ProcessState.ExitCode() != 0 {
+		return []byte{}, errors.Wrap(err, "Benchmark ended with an error.")
+	}
+	return data, err
+}
+
+func newClient(token string) *github.Client {
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(context.Background(), ts)
+	clt := github.NewClient(tc)
+	return clt
+}
+
+func formatFinalComment(rawTable string) string {
+	tableContent := strings.Split(rawTable, "\n")
+	for i := 0; i <= len(tableContent)-1; i++ {
+		e := tableContent[i]
+		switch {
+		case e == "":
+
+		case strings.Contains(e, "old ns/op"):
+			e = "| Benchmark | Old ns/op | New ns/op | Delta |"
+			tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+		case strings.Contains(e, "old MB/s"):
+			e = "| Benchmark | Old MB/s | New MB/s | Speedup |"
+			tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+		case strings.Contains(e, "old allocs"):
+			e = "| Benchmark | Old allocs | New allocs | Delta |"
+			tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+		case strings.Contains(e, "old bytes"):
+			e = "| Benchmark | Old bytes | New bytes | Delta |"
+			tableContent = append(tableContent[:i+1], append([]string{"|-|-|-|-|"}, tableContent[i+1:]...)...)
+
+		default:
+			// Replace spaces with "|".
+			e = strings.Join(strings.Fields(e), "|")
+		}
+		tableContent[i] = e
+	}
+	return strings.Join(tableContent, "\n")
+
+}
+
+func readFile(name string) (string, error) {
+	data, err := ioutil.ReadFile(name)
+	return string(data), err
+
+}
+
+func execCommand(command ...string) (*exec.Cmd, error) {
```

Comment:
why not refactor it to return the output and the exit code and use it everywhere?